x/stakeibc/module_ibc.go

```diff
@@ -184,6 +186,41 @@ func (im IBCModule) OnChanCloseConfirm(
 	portID,
 	channelID string,
 ) error {
+	im.keeper.Logger(ctx).Info(fmt.Sprintf("OnChanCloseConfirm: portID %s, channelID %s", portID, channelID))
+	controllerConnectionId, err := im.keeper.GetConnectionId(ctx, portID)
+	if err != nil {
+		ctx.Logger().Error(fmt.Sprintf("Unable to get connection for port: %s", portID))
+	}
+	// get host chain id from connection
+	hostChainId, err := im.keeper.GetChainID(ctx, controllerConnectionId)
+	if err != nil {
+		ctx.Logger().Error(fmt.Sprintf("Unable to obtain counterparty chain for connection: %s, port: %s, err: %s", controllerConnectionId, portID, err.Error()))
+		return nil
+	}
+	delegationAddress, err := icatypes.NewControllerPortID(types.FormatICAAccountOwner(hostChainId, types.ICAAccountType_DELEGATION))
+	if err != nil {
+		return err
+	}
+	if portID != delegationAddress {
```

Comment:
superrr nit, but might be a tad clearer if you changed this to an equality and nested everything else below - considering this is still supposed to be a generic OnChanClose function, and the delegation account closing is just one possible branch

```go
if portID == delegationAddress {
    ...
}
```